site/blog/ecs-simplified.mdx

```diff
@@ -0,0 +1,112 @@
+  - In the next step, we'll automatically build an image for the code in Github repo and then push it to this URI (all through SQL and using another high-level function named `ecr_build`).
+```sql
+SELECT ecr_build(
+  'https://github.com/iasql/iasql-engine/', -- the Github repo URL
+  (SELECT id
+    FROM repository
+    WHERE repository_name = 'simple-express-repository')::varchar(255), -- ECR repo for the image to be pushed
+  './examples/ecs-fargate/prisma/app', -- the subdirectory in Github repo
+  'main',  -- the Github branch or ref
+  NULL -- Github personal access token - can be omitted if public repository
+);
+```
+- This command tells IaSQL to go clone `iasql-engine` repository, build an image on the subdirectory specified, and then push it to the ECR repository created earlier by `ecs_simplified` module. Running the above command will automatically create a CodeBuild project and the related roles, etc. Then it'll start a build, and after it's successful all the created resources are deleted to ensure there won't be any additional charges to your AWS account.
+- To access your app on the cloud, get the load balancer address and use your browser to access the live version of it:
+```sql
+SELECT load_balancer_dns
+FROM ecs_simplified
+WHERE app_name = 'simple-express';
+```
+Then you can check if the server is running on the `url:8088/health` address.
+
+## Low-level Access to Resources
+
+So the `aws_ecs_simplified` module simplifies things, right? But what if you still need the level of control you had when you were doing all the steps manually? The traditional PaaS trade-off is that you can't grow your app beyond limitations as you don't have access to all the small details. The IaSQL approach is not limited in that way.
+
+Let's say you want your ECS container to be able to use the AWS CLI to provision an EC2 instance, and for that purpose its IAM role needs `AmazonEC2FullAccess` policy to work properly. `aws_ecs_simplified` does not have a column to configure such a thing, but that doesn't mean we're stuck.
+
+The good news is that you still have the full control over all resources in the deepest details. Let's fix your app's IAM role access by attaching the needed policy to its IAM role:
+```sql
+UPDATE iam_role
+SET attached_policies_arns = attached_policies_arns ||
+                             'arn:aws:iam::aws:policy/AmazonEC2FullAccess' -- attached_policies_arns is of text[] type
+WHERE role_name = 'simple-express-ecs-task-exec-role';
+
+```
+
+You want additional rules for the container's security group? No problem! just write the SQL and execute it, and it will be applied to the cloud within seconds. You want 3 copies of your container to be kept running with a round-robin load balancing on them? It's already there, just do an `UPDATE ecs_simplified SET desired_count = 3 WHERE app_name = 'simple_express';` and it's there for you.
```

Comment:
```suggestion
You want additional rules for the container's security group? No problem! Just write the SQL and execute it, and it will be applied to the cloud within seconds. You want 3 copies of your container to be kept running with a round-robin load balancing on them? It's already there, just do an `UPDATE ecs_simplified SET desired_count = 3 WHERE app_name = 'simple_express';` and it's there for you.
```

Reply:
Also, maybe this instead:

```suggestion
You want additional rules for the container's security group? No problem! Just write the SQL, execute it, and `SELECT iasql_apply()` it and it will be applied to the cloud within seconds. You want 3 copies of your container to be kept running with a round-robin load balancing on them? It's already there, just do an `UPDATE ecs_simplified SET desired_count = 3 WHERE app_name = 'simple_express';`, `SELECT iasql_apply()` it and it's there for you.
```